### PR TITLE
Make the websocket client reuse the client's certificate

### PR DESCRIPTION
**Pull request by pstch**
This change allows to use the `events()` on remote clients, by passing the SSL certificate and keyfile from the HTTP client to to the websocket client.

I had to move the definition of `self.cert` to pick up the default certificates.

This fixes issue #381.

### CODE REVIEW

**Comment by Mohamedemad4** [comment]
@stgraber This PR fixes an issue I am having and currently it's not merged. is there a reason/fix i can apply to this code so it can be merged upstream?

**Comment by stgraber** [comment]
@albertodonato

**Review by albertodonato — CHANGES_REQUESTED** [review]
Change looks good, thanks for the PR.

I just have a small nit inline.

Also, could you please rebase on current master and fix the conflict?

File: pylxd/client.py
Comment (pstch) [reply pylxd/client.py]: Done!

**Comment by pstch** [comment]
@albertodonato 

Rebased and inlined the if-statement.

However, I won't be able to test these changes for at least a few days due to some personal issues. 

I'd be glad if someone could just do a quick test before merging.

**Comment by Mohamedemad4** [comment]
I tested the changes locally and everything works great, @albertodonato  think we can merge?

**Comment by albertodonato** [comment]
Thanks for the changes.

I've made CI run on the PR, and some tests also need adjusting for the websocket call change.

Also, you'll need to sign-off your commits to make the DCO check pass.

I'll merge once those issues are addressed.

**Comment by pstch** [comment]
@albertodonato Oh ! I'm very sorry, I did the rebase too quickly and completely forgot about the tests. I will address that and sign the commits.

**Comment by pstch** [comment]
I will investigate the failed CI run this evening.

**Review by albertodonato — COMMENTED** [review]
It seems there is still a lint error, and a couple of test failures.
Could you please check those too?

Also, a minor comment inline

File: pylxd/tests/test_client.py
Comment (pstch) [reply pylxd/tests/test_client.py:330]: Will do.

**Comment by pstch** [comment]
Should be good now.

Tests were failing because the default certificates do not exist on the CI runner. I bypassed this check by forcing `Client` to use `DEFAULT_CERTS`. Since we're not testing this aspect of the `Client` class in this test, I think this is okay.

**Comment by albertodonato** [comment]
did you perhaps forget to push some changes?

**Comment by pstch** [comment]
I did. Sorry for that. Just pushed it.

**Review by albertodonato — APPROVED** [review]
+1, thanks!

**Pull request merged**
